frontend/public/components/configmap-and-secret-data.jsx

```diff
@@ -1,15 +1,56 @@
 import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import { Heading } from './utils';
 
-const ConfigMapAndSecretData = ({data, decode}) => {
-  decode = decode || (v => v);
-
+export const ConfigMapData = ({data}) => {
   const dl = [];
   Object.keys(data || []).sort().forEach(k => {
     dl.push(<dt key={`${k}-k`}>{k}</dt>);
-    dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{decode(data[k])}</pre></dd>);
+    dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{data[k]}</pre></dd>);
   });
 
   return <dl>{dl}</dl>;
 };
 
-export default ConfigMapAndSecretData;
+export class SecretData extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = { showSecret: false };
+    this.toggleSecret = this.toggleSecret.bind(this);
+  }
+
+  toggleSecret() {
+    this.setState({ showSecret: !this.state.showSecret });
+  }
+
+  render() {
+    const { data, decode } = this.props;
+    const { showSecret } = this.state;
+    const dl = [];
+    Object.keys(data).sort().forEach(k => {
+      dl.push(<dt key={`${k}-k`}>{k}</dt>);
+      dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{showSecret ? decode(data[k]) : '*****'}</pre></dd>);
+    });
+    return <React.Fragment>
+      <Heading text="Data" >
+        <a className="btn btn-link" onClick={this.toggleSecret}>{showSecret ? 'Hide Secret' : 'Reveal Secret'}</a>
+      </Heading>
+      <dl>{dl}</dl>
+    </React.Fragment>;
+  }
+}
+
+SecretData.propTypes = {
+  data: PropTypes.object,
+  decode: PropTypes.func
+};
+
+SecretData.defaultProps = {
+  data: [],
```

Comment:
propTypes specifies that data is an object, and we're calling `Object.keys()` on data, but it's initialized to an array?

Reply:
yeah, should be probably an empty object. `Object.keys(data || [])` confused me, but we want to update it there as well

Reply:
yeah, we want to default to an object. `Object.keys(data || {})` confused me... will update that as well